src/doc_table.h

```diff
@@ -77,11 +77,20 @@ typedef struct {
 } DocTable;
 
 /* increasing the ref count of the given dmd */
-#define DMD_Incref(md)                                                       \
-  if (md) {                                                                  \
-    RS_LOG_ASSERT(md->ref_count < (1 << 16), "overflow of dmd ref_count");   \
-    ++md->ref_count;                                                         \
-  }
+/*
+ * If the ref_count was 0 before we increased it, someone else already freed it,
```

Comment:
if someone freed this one, isn't it an invalid access?

Reply:
You're right. It won't work the way this comment explains without modifying the access to the pointer (AKA using a mutex).
On the other hand, this won't be a problem as this macro is used only if the spec's R/W lock is locked and therefore we either have only a single writer and no multi-threading or multi readers and then the spec's reference itself won't let the dmd get freed.
I'll fix the comment and move the macro